app/src/main/java/com/example/reduxtestapp/data/repository/country/CountryRepositoryImpl.kt

```diff
@@ -33,9 +39,13 @@ class CountryRepositoryImpl (
     override suspend fun getCountries(query: String): Either<ErrorState, List<CountryDto>> {
         return Either.catch {
             if (query.isNotEmpty()) {
-                countriesApi.searchCountries(query)
+                cacheManager.useCache(query) {
+                    countriesApi.searchCountries(query)
+                }
```

Comment:
create a cache key out of `query` and use it here (could be as simple as "search_countries_$query")